src/core/terrainGeneration/mapGenerator.js

```diff
@@ -0,0 +1,104 @@
+import { LCG } from "./LCG";
+import { Graphics } from "pixi.js";
+
+export class MapGenerator {
+    constructor(app, seed) {
+        this.app = app;
+        this.terrain = [];
+        // if no argument is passed into mapgenerator then it will just create a random terrain
+        if (seed == undefined) {
+            this.LCG = new LCG();
+        } else {
+            this.LCG = new LCG(seed);
+        }
+    }
+
+    generatePerlinNoiseLayer(amplitude, wavelength, canvasWidth) {
+        // amplitude tells us how high the peaks should be
+        // wavelength tells us how far adjacent troughs should be to one another
+        // this will be used to dictate the size of a single Perlin segment betweeen points A and points B
+        let x = 0;
+        let pseudoRandomNumberGenerator = this.LCG;
+        let pointA = pseudoRandomNumberGenerator.next();
```

Comment:
Is the random generator reliable for map building. Do you know how you plan to overlay the pixi sprite on top of it?
This comment is part of a review (different from the other comment) so reply to this one too to see if these ones work.

Reply:
I'm not sure.. All I know is that it's the simplest implementation of a pseudo random number generator, and it seems to work pretty well (it was in the article, other pseudo random number generators I think are a lot more advanced and too much math)

Reply:
I honestly haven't thought about the pixi sprite and actual game logic yet for the collision via plankjs, I'll try and figure it out tho. I'll try and send the article I used if you also want to look into it and try to understand it, it's pretty cool.

Reply:
https://codepen.io/OliverBalfour/post/procedural-generation-part-1-1d-perlin-noise this the article I found, it took me so long cus there was barely any info on 1D perlin noise generation, most of it was either 2d or 3d.

Reply:
Will give this a read, looks interesting. Can't wait to work on this more during the break